### ISSUE TITLE
Introduce in-Textarea highlighting

### ISSUE THREAD

**Issue by garrett-eclipse** [Mte90/GlotDict#162]
This ticket will house the efforts for introducing in-Textarea highlighting for spaces, terms and other items.

Originally the idea stems from this GlotPress issue;
https://github.com/GlotPress/GlotPress-WP/issues/863

This is quite an involved implementation as we'd probably have to retrofit the textarea with a complimentary div which visually replaces the textarea to support highlighting.
We could potentially use a library like this - https://codersblock.com/blog/highlight-text-inside-a-textarea/

Some ideas for what should be highlighted in the text area are;
- highlight ending space
- highlight non-breaking space
- highlight glossary terms with click replace
- highlight … with click replace
- etc

In some cases such as the term highlight and the trailing ... highlight we can take advantage of the 'Fix Warning' functionality being built for 1.5 in #145 so that on-click those highlights can execute replace logic.

**Comment by garrett-eclipse** [comment]
We may want to look at the Grammarly extension to get some ideas as they have a really nice in-textarea highlight with tooltip that contains replace actions.
https://dy6j70a9vs3v1.cloudfront.net/funnel_wap/static/files/c27b841c0710f8bd1d8e43c76a111e77/card_correct.png

**Comment by garrett-eclipse** [comment]
There's also the 'contenteditable' path as mentioned on GP ticket https://github.com/GlotPress/GlotPress-WP/issues/166

**Comment by Mte90** [comment]
If we can use a library is better to have less custom code untested a lot but we will find something :-)

**Comment by garrett-eclipse** [comment]
Agreed a library is going to be best, hopefully something lightweight.
We'll want to avoid WYSIWYG editors and maybe contenteditable now that I look at it more since both support injecting markup (Bold, Italics, etc) which affects the integrity of the string as we shouldn't be injecting markup that's not already present.

**Comment by psmits1567** [comment]
I propose we close this issue, as GlotPress now marks the start and end within the original
